Source/WebGPU/WebGPU/PresentationContextIOSurface.mm

```diff
@@ -105,6 +105,12 @@
     MTLTextureDescriptor *textureDescriptor = [MTLTextureDescriptor texture2DDescriptorWithPixelFormat:Texture::pixelFormat(effectiveFormat) width:width height:height mipmapped:NO];
     textureDescriptor.usage = Texture::usage(descriptor.usage, effectiveFormat);
     bool needsLuminanceClampFunction = false;
+    for (IOSurface *iosurface in m_ioSurfaces) {
+        if (iosurface.height < 0 || height * textureDescriptor.arrayLength > static_cast<NSUInteger>(iosurface.height)) {
```

Comment:
should we also add: `iosurface.height != height || iosurface.width != width` to this if check?

And isn't `textureDescriptor.arrayLength` always 1?

Reply:
I like your heights-and-widths-must-be-equal suggestion.  I think arrayLength is always 1, but I'm not sure.